screens/VehicleDetailsScreen.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from "react";
+import { ActivityIndicator, StyleSheet } from "react-native";
+import { Text, View } from "../components/Themed";
+import { Vehicle } from "../models";
+import fetchAPI from "../fetchApi";
+
+interface DetailsItemProps {
+  name: string;
+  value: string;
+}
+
+const DetailsItem = ({ name, value }: DetailsItemProps) => (
+  <Text style={styles.listItem}>
+    {name}: {value}
+  </Text>
+);
+
+const emptyVehicle = (): Vehicle => ({
+  name: "---",
+  model: "---",
+  manufacturer: "",
+  cost_in_credits: "",
+  max_atmosphering_speed: "",
+  passengers: "",
+});
+
+export default function VehicleDetailsScreen({ navigation, route }) {
+  const [isLoading, setLoading] = useState(true);
+  const [data, setData] = useState<Vehicle>(emptyVehicle);
+
+  useEffect(() => {
+    async function loadVehicle() {
+      let id: number = route.params.id;
+
+      const json = await fetchAPI.getVehicle(id);
+      setData(json);
+    }
+    loadVehicle();
+    setLoading(false);
+  }, []);
+
+  return (
+    <View style={styles.container}>
+      <Text style={styles.title}>Wehicle details:</Text>
+      <View
+        style={styles.separator}
+        lightColor="#eee"
+        darkColor="rgba(255,255,255,0.1)"
+      />
+
+      {isLoading ? (
```

Comment:
Być może tak naprawdę nie potrzebujesz `isLoading` -> wystarczy sprawdzić czy `data` nie jest puste. 
TS ma taką fajną umiejętność że jeśli `zmienna` jest typu `Costam | undefined` to po sprawdzeniu `if(zmienna) {` , `zmienna` już jest tylko `Costam`